src/modules/email/persistence/email.repo.ts

```diff
@@ -153,6 +169,41 @@ export class EmailRepo {
         return new PersonEmailResponse(emailAddresses[0].status, emailAddresses[0].address);
     }
 
+    /**
+     * Returns a map with key-value pairs personId -> PersonEmailResponse.
+     * Only enabled addresses will be part of the response, so the resulting map can return UNDEFINED for a personId
+     * when either no email-addresses could be found for that person or only email-addresses with status other than ENABLED.
+     * @param personIds
+     */
+    public async getEmailAddressAndStatusForPersonIds(
+        personIds: PersonID[],
+    ): Promise<Map<PersonID, PersonEmailResponse>> {
+        const addresses: EmailAddress<true>[] = await this.findByPersonIdsSortedByUpdatedAtDesc(personIds);
+        const enabledAddresses: EmailAddress<true>[] = addresses.filter(
+            (ea: EmailAddress<true>) => ea.status === EmailAddressStatus.ENABLED,
+        );
+        const responseMap: Map<PersonID, PersonEmailResponse> = new Map<PersonID, PersonEmailResponse>();
+        let enabledAddressesForPersonId: EmailAddress<true>[];
+        personIds.map((personId: PersonID) => {
```

Comment:
Deutlich effizienter wäre es, die Adressen zu laden sortiert nach `personId` und `updatedAt`.
Dann kann man linear durch die Adressen gehen und in die Map einpflegen.